tests/utils/test_url.py

```diff
@@ -0,0 +1,90 @@
+import unittest
+from unittest.mock import MagicMock, patch
+
+from page_analyzer.utils.url import (
+    check_url,
+    normalize,
+    parse_page_data,
+    validate,
+)
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {}
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    expected_errors = {"name": "Некорректный URL"}
+    assert errors == expected_errors
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    expected_url = "http://www.example.com"
+    assert normalized_url == expected_url
+
+
+class TestCheckUrl(unittest.TestCase):
```

Comment:
it's not necessary to base your tests on `unittest.TestCase` class if you use patch or mock from `unittest`
tests in pytest are usually functions, not classes